karrot/subscriptions/receivers.py

```diff
@@ -244,15 +244,44 @@ def send_group_membership_updates(sender, instance, created, **kwargs):
         # notification types are only visible to one user
         send_group_detail(group, user=membership.user)
 
+
+@receiver(post_save, sender=GroupMembership)
+@on_transaction_commit
```

Comment:
Is it needed to add this?

Reply:
It seems conceptually necessary to prevent a race condition - adding members is part of a larger transaction (usually accepting an application), and the frontend would try to reload the user list immediately after receiving the websocket event. If the frontend reloads before the transaction commits, it would receive the outdated list.

Reply:
It seems conceptually necessary to prevent a race condition - adding members is part of a larger transaction (usually accepting an application), and the frontend would try to reload the user list immediately after receiving the websocket event. If the frontend reloads before the transaction commits, it would receive the outdated list.

With this, `application.accept()` is already wrapped in `@transaction.atomic`, so it should only send out the websocket event after the transaction has been committed.

Reply:
Yup, seems good. I actually wondered if it's best to _always_ have the receivers `@on_transaction_commit`. I think at some point I played around with a custom annotation to do this, and also run them in huey jobs, as some potentially are slow. Anyway, that's another story :)